src/rgw/rgw_s3_filter.h

```diff
@@ -61,16 +67,25 @@ struct rgw_s3_key_value_filter {
 
   void encode(bufferlist& bl) const {
     ENCODE_START(2, 1, bl);
-      encode(kv, bl);
-      encode(negative_filter_map, bl);
+    if(struct_v >= 2)
```

Comment:
i hope this is ok, @yuvalif.

Reply:
```cpp
KeyValueMap kv;
for (const auto& it : kvt) {
  if (it.second.second) {
    kv.emplace(it.first, it.second.first)
  }
}
encode(kv, bl);
encode(kvt, bl);
```
* when you encode you don't know what other RGW versions exist i nthe cluster
* you have to encode according to the data member order in the struct